Unparser drops line continuation on #pragma rose (outline becomes code)

Summary
- The unparser drops line continuation on a `#pragma rose` directive, emitting the continuation token as a standalone identifier and breaking the backend compile.

Repro
- `ctest --test-dir build -R '^frontend_integration_c_test2010_03\.c$' -V`
- Observed in `build/tests/nonsmoke/functional/CompileTests/frontend_integration/rose_test2010_03.c`:
  - Original:
    ```
    #pragma rose \
      outline
    ```
  - Unparsed:
    ```
    #pragma rose
      outline
    ```
  - Backend error: `use of undeclared identifier 'outline'`.

Root cause
- Token-based unparse loses the line-continuation backslash in preprocessing info for multi-line pragmas.

Long-term fix (no workarounds)
- Preserve the `\` line-continuation token for `#pragma` directives in the token stream, or
- Canonicalize multi-line pragmas into a single-line `#pragma rose outline` at unparse time.
- Add a dedicated test for multi-line pragmas with continuations.

`ctest --test-dir build -j16 -L C_tests_failing --output-on-failure` shows `test2010_03.c` failing during unparse/compile:

```
rose_test2010_03.c:15:5: error: use of undeclared identifier 'outline'
```

This is the specimen with `#pragma rose \` followed by `outline` on the next line. It looks like the line continuation is being dropped, so `outline` becomes a standalone token in the output. Log: `/tmp/CompileTests_C_tests_failing.log`.

--- a/tests/nonsmoke/functional/CompileTests/C_tests/CMakeLists.txt
+++ b/tests/nonsmoke/functional/CompileTests/C_tests/CMakeLists.txt
@@ -882,7 +882,6 @@ set(TESTCODE_CURRENTLY_FAILING
   test2009_06.c
   test2009_19.c
   test2010_02.c
-  test2010_03.c
   test2010_04.c
   test2010_05.c
   test2010_12.c

